### PR TITLE
Fix things metioned in #11

### PR DESCRIPTION
**Pull request by bdke**
Thx for telling me those mistakes
Hopefully nothing goes wrong after this commit 👍

### CODE REVIEW

File: src/jmc/compile/command/builtin_function/jmc_command.py [hunk 1]
```diff
@@ -623,136 +622,52 @@ class AdvancementRevoke(JMCFunction):
     def call(self) -> str:
         advancement = self.args["advancement"]
         target = self.args["target"]
-        selection = self.args["selection"]
+        type = self.args["type"]
```
Comment (WingedSeal) [inline src/jmc/compile/command/builtin_function/jmc_command.py:625]: `type` is overriding built-in `type` of python
name should be changed to `type_` or something else

File: src/jmc/compile/datapack_data.py [hunk 2]
```diff
@@ -6,6 +6,91 @@
 if TYPE_CHECKING:
     from ..compile.tokenizer import Token
 
+@dataclass
+class MinecraftAdvancements:
```
Comment (WingedSeal) [inline src/jmc/compile/datapack_data.py:10]: This is not what I meant you should define in `jmc_command.py` outside of class

`MINECRAFT_ADVANCEMENTS = {`

File: src/jmc/compile/datapack_data.py [hunk 3]
```diff
@@ -6,6 +6,91 @@
 if TYPE_CHECKING:
     from ..compile.tokenizer import Token
 
+@dataclass
```
Comment (WingedSeal) [inline src/jmc/compile/datapack_data.py:9]: Misuse of `@dataclass`

File: src/jmc/compile/command/builtin_function/jmc_command.py [hunk 4]
```diff
@@ -623,136 +622,52 @@ class AdvancementRevoke(JMCFunction):
...
-        ]
-        files = list(self.datapack.defined_file_pos.keys())
+
+        files = self.datapack.jsons.keys()
 
         if namespace == "minecraft":
-            minecraft_files: list[str] = ['/'.join(i.split("/")[2:]) for i in files + minecraft_advancements if i.split("/")[0] == "minecraft"]
-            if advancement not in minecraft_files:
+            if not(f"minecraft/advancements/{advancement}" in MinecraftAdvancements().advancements or f"minecraft/advancements/{advancement}" in files):
                 raise JMCFileNotFoundError(f"'{advancement}' advancement in '{namespace}' is not defined or missing")
```
Comment (WingedSeal) [inline src/jmc/compile/command/builtin_function/jmc_command.py:632]: Use `JMCValueError` instead

**Review by WingedSeal — CHANGES_REQUESTED** [review]
Almost there!

I also revert your last pull request so I'll have to deal with merge conflict T_T

File: src/jmc/compile/datapack_data.py (hunk 3 above)
Comment (bdke) [reply src/jmc/compile/datapack_data.py:9]: I thought it is sth for storing data
thx for telling me that

**Review by WingedSeal — APPROVED** [review]

**Pull request merged**